### PR TITLE
Feature/user registration

### PR DESCRIPTION
**Pull request by LoginALee**
## Explain why this change is being made.
- As a no registered user, I can sign up in the app through a button which redirects me to the signup page
- As a registered user, I would like to be able to sign in to the application so that I can track my habits constantly
## Explain how this is accomplished.
- Login:
  - Having already a registered account, you can log in into the app
- Sign up:
  - If you don't already have a registered account, you can sign up into the app
## How do you manually test this?
- Login:
  - Been at home page, click the Log in button, then you get redirected to the Log in page, where you need to fill in a form with user email and password, then you click the Log in button to get redirected the home page
- Sign up:
  - Been at home page, click the Sign up button, then you get redirected to the Sign up page, where you need to fill in a form with user fields (first name, last name, etc...), then you click the Sign up button to finish the sign up process and get redirected to the home page

## Screenshots
- Login:
image

- Sign up:
image

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+  validates :first_name, :last_name, :phone, :password, presence: true
+  validates :phone, length: { is: 10 }
```
Comment (jvargas98) [inline app/models/user.rb]: I think the country code is missing, the length should be longer.

File: spec/factories/users.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :user do
+    email { Faker::Internet.unique.safe_email }
+    password { Faker::Internet.password(min_length: 6) }
+    password_confirmation { password } 
+    first_name { Faker::Name.first_name }
+    last_name { Faker::Name.last_name }
+    phone { Faker::Number.number(digits: 10) }
```
Comment (jvargas98) [inline spec/factories/users.rb]: The same here, are you including the country prefix?

**Review by jvargas98 — APPROVED** [review]

**Review by GregVaz — APPROVED** [review]

File: db/migrate/20210329191131_devise_create_users.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
+
+      ## Lockable
+      # t.integer  :failed_attempts, default: 0, null: false # Only if lock strategy is :failed_attempts
+      # t.string   :unlock_token # Only if unlock strategy is :email or :both
+      # t.datetime :locked_at
```
Comment (OmauryMarmolejo) [inline db/migrate/20210329191131_devise_create_users.rb]: Is this code needed?

Comment (LoginALee) [reply db/migrate/20210329191131_devise_create_users.rb]: Not really, good observation :smile:

File: spec/views/home/index.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe "home/index.html.erb", type: :view do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (GregVaz) [inline spec/views/home/index.html.erb_spec.rb]: Is this test necessary? Because you are already using feature tests

**Review by OmauryMarmolejo — APPROVED** [review]

File: spec/views/home/index.html.erb_spec.rb (hunk 4 above)
Comment (LoginALee) [reply spec/views/home/index.html.erb_spec.rb]: I think its not necessary, thanks Goyo :rocket:

**Pull request merged**